source/common/upstream/health_discovery_service.cc

```diff
@@ -141,6 +143,9 @@ void HdsDelegate::onReceiveMessage(
   stats_.requests_.inc();
   ENVOY_LOG(debug, "New health check response message {} ", message->DebugString());
 
+  // Reset
```

Comment:
I'm trying to understand the behavior of the cluster and cluster manager when HdsClusters come and go. I don't see where we add and remove the cluster in cluster manager. This itself governs behavior like connection pooling, which are relevant to whether we want to always do a clear or if we should be smarter. Can you dig a bit and figure out how the HdsClusters are added to `ClusterManager`? Maybe they are not, in which case, how does HTTP and TCP connection pooling work? If it doesn't, that seems worth noting in a TODO at least.

Reply:
There is no connection pooling yet. I added a TODO.